sklearn/model_selection/_validation.py

```diff
@@ -113,12 +124,45 @@ def cross_val_score(estimator, X, y=None, groups=None, scoring=None, cv=None,
     --------
     >>> from sklearn import datasets, linear_model
     >>> from sklearn.model_selection import cross_val_score
+    >>> from sklearn.metrics.scorer import make_scorer
+    >>> from sklearn.metrics import confusion_matrix
+    >>> from sklearn.svm import LinearSVC
     >>> diabetes = datasets.load_diabetes()
     >>> X = diabetes.data[:150]
     >>> y = diabetes.target[:150]
     >>> lasso = linear_model.Lasso()
-    >>> print(cross_val_score(lasso, X, y))  # doctest: +ELLIPSIS
-    [ 0.33150734  0.08022311  0.03531764]
+
+    >>> print(cross_val_score(lasso, X, y))               # doctest: +ELLIPSIS
+    [ 0.33...  0.08...  0.03...]
+
+    >>> scores = cross_val_score(lasso, X, y,
+    ...                          scoring=('r2', 'neg_mean_squared_error'))
+    >>> print(scores['neg_mean_squared_error'])           # doctest: +ELLIPSIS
+    [-3635.5... -3573.3... -6114.7...]
+    >>> print(scores['r2'])                               # doctest: +ELLIPSIS
+    [ 0.33...  0.08...  0.03...]
+
+    >>> svm = LinearSVC(random_state=0)
+    >>> # A sample toy binary classification dataset
+    >>> X, y = datasets.make_classification(n_classes=2, random_state=0)
+
+    >>> tp = lambda y_true, y_pred: confusion_matrix(y_true, y_pred)[0, 0]
+    >>> tn = lambda y_true, y_pred: confusion_matrix(y_true, y_pred)[0, 0]
+    >>> fp = lambda y_true, y_pred: confusion_matrix(y_true, y_pred)[1, 0]
+    >>> fn = lambda y_true, y_pred: confusion_matrix(y_true, y_pred)[0, 1]
+
+    >>> # scoring can also be a dict mapping the scorer name to scorer callable
```

Comment:
If we're sure we want this example, please move this comment to the top of the example and alter it so it's clearly a heading. I don't think we want this many blank lines within one example case.